x-pack/plugins/viz_editor/public/pseudo_plugins/xy_chart_plugin/yaxis_editor.tsx

```diff
@@ -21,46 +20,38 @@ export function YAxisEditor({
   onChangeVisModel: (visModel: VisModel) => void;
 }) {
   const currentOperation: any = selectColumn(col, visModel) || { operation: 'count' };
-
-  const fieldName = 'argument' in currentOperation ? currentOperation.argument.field : '';
-
-  const field = visModel.datasource.fields.find((f: DatasourceField) => f.name === fieldName);
-
-  const onDropField = (droppedField: DatasourceField) => {
-    const operation = fieldToOperation(droppedField, getOperatorsForField(droppedField)[0]);
+  const fieldName = currentOperation.argument && currentOperation.argument.field;
+  const onDropField = (field: DatasourceField) => {
+    const operation = fieldName
+      ? { ...currentOperation, argument: { ...currentOperation.argument, field: field.name } }
+      : fieldToOperation(field, 'sum');
 
     onChangeVisModel(updateColumn(col, operation, visModel));
   };
 
-  const onChangeOperation = (e: React.ChangeEvent<HTMLSelectElement>) => {
-    if (!field) {
-      return;
-    }
-
-    const operationName = e.target.value as SelectOperator;
-    const operation = fieldToOperation(field, operationName);
-
-    onChangeVisModel(updateColumn(col, operation, visModel));
-  };
+  const column = selectColumn(col, visModel);
 
-  const options = field
-    ? getOperatorsForField(field).map(opName => ({
-        value: opName,
-        text: `${opName} of ${fieldName}`,
-      }))
-    : currentOperation;
+  if (!column) {
+    // TODO...
+    return <span>N/A</span>;
+  }
 
   return (
-    <Draggable canHandleDrop={(f: DatasourceField) => !!f} onDrop={onDropField}>
-      {field ? (
-        <EuiSelect
-          options={options}
-          value={currentOperation.operation}
-          onChange={onChangeOperation}
-        />
-      ) : (
-        columnSummary(selectColumn(col, visModel))
-      )}
+    <Draggable
+      canHandleDrop={(f: DatasourceField) => f && f.type === 'number'}
```

Comment:
Could there be an interaction between `canHandleDrop` and which operations are already configured?

Reply:
Good question, this basically comes down to whether we want this from a UX perspective. 

At the moment the `operation_editor` looks whether the dropped field is applicable to the currently configured operation and just switches out the field reference and keeps the currently configured params. If a field is dropped which is not compatible with the currently defined operation (e.g. there is a date histogram and you drop a string field), it overwrites the operation completely with the default operation and throws away the configured parameters. https://github.com/elastic/kibana/pull/34544/files#diff-6f0ad430eb014d21c54f631dd572474e

Your way (simply not allowing to drop the string field on a configured date histogram) is also valid, not sure what's better actually. Maybe asking the user what they want to do?